transport/tchannel/handler.go

```diff
@@ -152,6 +152,17 @@ func (h handler) callHandler(ctx context.Context, call inboundCall) error {
 		return err
 	}
 
+	// We recover panics from the user handler.
+	defer func() {
+		if r := recover(); r != nil {
+			if err, ok := r.(error); ok {
+				returnErr = err
+			} else {
+				returnErr = fmt.Errorf("%q", r)
+			}
+		}
+	}()
+
```

Comment:
Yes, but in a following PR. As we discussed, there is few error conversions from user handler to yarpc error that can be factored in a single place.